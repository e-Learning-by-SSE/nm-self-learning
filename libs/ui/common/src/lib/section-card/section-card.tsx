import { ReactNode } from "react";

export function SectionCard({ children, className }: { children: ReactNode; className?: string }) {
	return (
		<div
			className={`card relative grid w-full items-start rounded-lg border border-light-border bg-white ${
				className ?? ""
			}`}
		>
			{children}
		</div>
	);
}

export function SectionCardHeader({
	title,
	subtitle
}: {
	title: string;
	subtitle?: string | null;
}) {
	return (
		<div className="mb-8 grid items-start gap-2">
			<h2 className="text-2xl">{title}</h2>
			{subtitle && <span className="text-light">{subtitle}</span>}
		</div>
	);
}

export function SectionHeader({
	title,
	subtitle,
	button
}: {
	title: string;
	subtitle?: string | null;
	button?: ReactNode;
}) {
	return (
		<div className="mb-8 mt-4">
			<div className="flex items-center gap-4">
				<h2 className="flex-grow text-4xl">{title}</h2>

				{button && <div className="mr-0.5">{button}</div>}
			</div>
			{subtitle && <p className="mt-2 text-light">{subtitle}</p>}{" "}
		</div>
	);
}
