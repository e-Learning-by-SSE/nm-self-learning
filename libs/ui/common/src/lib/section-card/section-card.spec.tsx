import { render } from "@testing-library/react";
import { SectionCard } from "./section-card";

describe("SectionCard", () => {
	describe("Layout", () => {
		it("Default Layout", () => {
			const { container } = render(<SectionCard>Content</SectionCard>);
			const html = container.innerHTML;

			// Regression test: Checks that the component renders as in previous versions
			// To update snapshot run: npx jest -u <path/to/test/file.spec.ts>
			expect(html).toMatchSnapshot();
		});

		it("Custom Layout", () => {
			// Tests that layout can be customized (additional CSS classes added)
			const additionalClass = "border-dotted";
			const { container } = render(
				<SectionCard className={additionalClass}>Content</SectionCard>
			);
			const html = container.innerHTML;

			// Check for Key criteria: dashed border added
			expect(html).toContain(additionalClass);
		});
	});

	describe("Functionality", () => {
		it("Content", () => {
			const children = "Content";
			const { container } = render(<SectionCard>{children}</SectionCard>);

			// Only specified children was added
			expect(container.children.length).toBe(1);
			expect(container.children[0].innerHTML).toEqual(children);
		});
	});
});
