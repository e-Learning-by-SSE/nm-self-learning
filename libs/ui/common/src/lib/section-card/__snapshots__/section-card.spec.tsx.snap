// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionCard Layout Default Layout 1`] = `"<div class=\\"card relative grid w-full items-start rounded-lg border border-light-border bg-white \\">Content</div>"`;
